build/root/Makefile.generated_files

```diff
@@ -451,7 +451,7 @@ $(OPENAPI_OUTFILE): $(OPENAPI_GEN) $(KNOWN_VIOLATION_FILENAME)
 	    -h $(BOILERPLATE_FILENAME)                                                  \
 	    -r $(REPORT_FILENAME)                                                       \
 	    "$$@";                                                                      \
-	diff $(REPORT_FILENAME) $(KNOWN_VIOLATION_FILENAME) ||                          \
+	diff <(sort $(REPORT_FILENAME)) <(sort $(KNOWN_VIOLATION_FILENAME)) ||          \
```

Comment:
is the output from the tool not already sorted?

Reply:
The output from the tool is not sorted yet.

https://github.com/kubernetes/kube-openapi/blob/e3762e86a74c878ffed47484592986685639c2cd/pkg/generators/api_linter.go#L91-L95

It's not hard to add sorting in the tool though

Reply:
my (mild) preference would be to have the output from the tool be sorted, as the checked-in violations file already is sorted (or probably should be).

Reply:
Yes I agree. I'll update the tool to have the output sorted. 

I'm also planning on adding a `hack/update-api-rules-violation-exceptions.sh` script which does a dry-run of the tool, so people can get the (sorted) expected output instead of manually modifying the `violation_exceptions.list`.